cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -277,7 +277,6 @@ func Run(c *config.CompletedConfig, stopCh <-chan struct{}) error {
 
 	// Start the main lock
 	go leaderElectAndRun(c, id, electionChecker,
-		c.ComponentConfig.Generic.LeaderElection.ResourceLock,
```

Comment:
keep using/plumbing `LeaderElection.ResourceLock`, and add validation in GenericControllerManagerConfigurationOptions#Validate that if LeaderElection.ResourceLock is set, it is set to `leases`